pkg/api/v1/types.go

```diff
@@ -675,6 +675,7 @@ type HostPathVolumeSource struct {
 	// Path of the directory on the host.
 	// More info: https://kubernetes.io/docs/concepts/storage/volumes#hostpath
 	Path string `json:"path" protobuf:"bytes,1,opt,name=path"`
+	Type string `json:"type" protobuf:"bytes,2,opt,name=type"`
```

Comment:
We should define a type for this, `HostPathType` and define a set of enumerated strings with comments explaining their meanings.